internal/server/event_server.go

```diff
@@ -233,9 +233,11 @@ func eventKeyFunc(r *http.Request) (string, error) {
 
 	objectGroup := event.InvolvedObject.GetObjectKind().GroupVersionKind().Group
 
+	originRevisionKey := fmt.Sprintf("%s/%s", objectGroup, eventv1.MetaOriginRevisionKey)
 	revisionKey := fmt.Sprintf("%s/%s", objectGroup, eventv1.MetaRevisionKey)
-	revision, ok := event.Metadata[revisionKey]
-	if ok {
+	if originRevision, ok := event.Metadata[originRevisionKey]; ok {
+		comps = append(comps, "revision="+originRevision)
+	} else if revision, ok := event.Metadata[revisionKey]; ok {
```

Comment:
We should add both to the rate limiter key, the origin revision may be the same while the OCI digest may change.

Reply:
fixed, thanks